assets/javascripts/modules/autoCompleteFactory.js

```diff
@@ -2,15 +2,37 @@ require('jquery');
 var autoComplete = require('./autoComplete.js');
 
 var createAutoCompleteCountries = function () {
-  var $chooseCountryAutoCompleteElem = $('.js-choose-country-auto-complete');
-  var $countryCodeInput = $('#countryCode');
-  var suggestionDisplayTemplate = function (title, value) {
-    return title + ' (+' + value + ')';
-  };
+  $('.js-choose-country-auto-complete').each(function(){
+    
+    var $countryCodeInput;
+    if($('#countryCode').length > 0){
+      $countryCodeInput = $('#countryCode');
+    } else {
+      $countryCodeInput = $('[id="' + $(this).attr("data-suggestions") + '"]');    
+    }
```

Comment:
I wanted to allow $countryCodeInput to be definable from the view as it is currently hard-coded in the factory. If the ID is already defined in the view (or the list is not one of countries) we need to allow this (using a defined ID also restricts instance use).

Agree that if data-suggestions is used elsewhere for validation I need to separate out my added functionality to make it clearer - didn't realise that integration was there. I had done this originally but decided to re-use the data-suggestions attribute as it seemed un-used.